src/lib/operators/aggregate_sort.cpp

```diff
@@ -0,0 +1,416 @@
+          auto aggregate_function =
+              AggregateFunctionBuilder<ColumnDataType, AggregateType, AggregateFunction::CountDistinct>()
+                  .get_aggregate_function();
+          _aggregate_values<ColumnDataType, AggregateType, AggregateFunction::CountDistinct>(
+              aggregate_group_offsets, aggregate_index, aggregate_function, sorted_table);
+          break;
+        }
+      }
+    });
+
+    aggregate_index++;
+  }
+
+  // append output to result table
+  result_table->append_chunk(_output_segments);
+
+  return result_table;
+}
+
+std::shared_ptr<AbstractOperator> AggregateSort::_on_deep_copy(
+    const std::shared_ptr<AbstractOperator>& copied_input_left,
+    const std::shared_ptr<AbstractOperator>& copied_input_right) const {
+  return std::make_shared<AggregateSort>(copied_input_left, _aggregates, _groupby_column_ids);
+}
+
+void AggregateSort::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
+
+void AggregateSort::_on_cleanup() {}
+
+template <typename ColumnType>
+void AggregateSort::_write_aggregate_output(boost::hana::basic_type<ColumnType> type, ColumnID column_index,
+                                            AggregateFunction function) {
+  switch (function) {
+    case AggregateFunction::Min:
+      write_aggregate_output<ColumnType, AggregateFunction::Min>(column_index);
+      break;
+    case AggregateFunction::Max:
+      write_aggregate_output<ColumnType, AggregateFunction::Max>(column_index);
+      break;
+    case AggregateFunction::Sum:
+      write_aggregate_output<ColumnType, AggregateFunction::Sum>(column_index);
+      break;
+    case AggregateFunction::Avg:
+      write_aggregate_output<ColumnType, AggregateFunction::Avg>(column_index);
+      break;
+    case AggregateFunction::Count:
+      write_aggregate_output<ColumnType, AggregateFunction::Count>(column_index);
+      break;
+    case AggregateFunction::CountDistinct:
+      write_aggregate_output<ColumnType, AggregateFunction::CountDistinct>(column_index);
+      break;
+  }
+}
+
+template <typename ColumnType, AggregateFunction function>
+void AggregateSort::write_aggregate_output(ColumnID column_index) {
+  // retrieve type information from the aggregation traits
+  auto aggregate_data_type = AggregateTraits<ColumnType, function>::AGGREGATE_DATA_TYPE;
+
+  const auto& aggregate = _aggregates[column_index];
+
+  if (aggregate_data_type == DataType::Null) {
+    // if not specified, it’s the input column’s type
+    aggregate_data_type = input_table_left()->column_data_type(*aggregate.column);
+  }
+
+  // Generate column name, TODO(anybody), actually, the AggregateExpression can do this, but the Aggregate operator
+  // doesn't use Expressions, yet
+  std::stringstream column_name_stream;
+  if (aggregate.function == AggregateFunction::CountDistinct) {
+    column_name_stream << "COUNT(DISTINCT ";
```

Comment:
We copied this code from the hash aggregate.

I'm not really sure what you mean, `aggregate_function_to_string` is a `boost::bimap<AggregateFunction, std::string>`, not an output stream?
What would be the "original place" for you? Something like 
`column_name_stream << "%s(%s)".format(aggregate_name, column_name);`

Reply:
I have thought of something like sprintf: http://www.cplusplus.com/reference/cstdio/sprintf/
I would prefer a formatted naming function that already includes a placeholder for (
For example: COUNT%sDISTINCT and SUM%s

Reply:
c++ doesn't have built-in string formatting as, e.g., python does, sadly. Using sprintf is out of the question because it is cumbersome to use properly. The way the string is formatted here is fine